src/algos/asymptotic_analysis/prework/analysis.clj

```diff
@@ -0,0 +1,88 @@
+
+(defn arithmetic-sum
+  [n]
+  (/ (* n (inc n)) 2))
+; PROFILING:
+;algos.core=> (map #(time (sum-to-n (* % 1000000))) (range 1 11))
+;"Elapsed time: 44.732251 msecs"
+;"Elapsed time: 61.726361 msecs"
+;"Elapsed time: 90.809637 msecs"
+;"Elapsed time: 119.782373 msecs"
+;"Elapsed time: 148.348564 msecs"
+;"Elapsed time: 174.791564 msecs"
+;"Elapsed time: 202.473209 msecs"
+;"Elapsed time: 227.27636 msecs"
+;"Elapsed time: 254.373277 msecs"
+;"Elapsed time: 282.996177 msecs"
+; O(1) bc takes same amt of time regardless of input
+; math == mind-blown
+
+
+;1. Understand the problem
+; Do the given strings have the same frequency of letters in a different order
+; "assume that the two strings in question use symbols from the set of 26 lowercase alphabetic characters"
+;
+;2. Devise a plan
+; Clojure has a [`frequencies` fn](https://clojuredocs.org/clojure.core/frequencies) which will give the frequency of
+; the characters in a string. This would tell me if they had the same characters but I'd need to do more work to find out
+; if they appear in the same order.
+; Other benefits to 'preprocessing' the chars in the string would be:
+;- equality comparison which would tell us upfront whether they are the same letters in same order
+;- length comparison which would tell us whether they have the same # of letters
+; Rather than reading both strings entirely and then performing *more* work, I could loop through one string and:
+; - see if each letter is the present in the other exiting if one isn't
+; - keeping track of whether all the letters are in the same position
+; based on the pangram solutions there may be a way of solving this using bits but I don't know it...yet ;)
+;
+; 3. Carry out the plan
+(defn anagram?
+  [s1 s2]
+  (cond
+    (not (= (count s1) (count s2))) false
+    (= s1 s2) false
+    :else (= (frequencies s1) (frequencies s2))))
+
+; 4. Look back
+;After looking at the 4 example implementations I am happy with mine!
+;Mine takes into account the edge case of if the two strings have the same letters in the same order (i.e. are the same string)
+;Also after reading about performance of Python types I decided to find out the same for Clojure to evaluate my anagram approach:
+; count -> O(1), Clojure strings are Java Strings which store the size on creation as a separate field
+; equality -> O(N), uses Java's .equals() which is O(1) in a few cases (ex. if strings are different lengths) but worst-case
+; might have to compare each character
+; frequencies ->
+; frequencies is really interesting [under the hood](https://github.com/clojure/clojure/blob/841fa60b41bc74367fb16ec65d025ea5bde7a617/src/clj/clojure/core.clj#L7123)!
+; `get` on a string is O(1) and always returns nil so the fact that yr even checking for the letter is what increments
+; the frequency count.
+; `assoc` is O(log N) thought there is [an argument here](https://news.ycombinator.com/item?id=8503912) that I'm too new to
+; have a solid opinion on.
```

Comment:
based on all this my solution is Log Linear?

; O(1) -> if strings of unequal length
; O(N) -> if same string
; otherwise O(N log N) -> 1*N (get) + log N * N (assoc) + N (reduce) x 2 (bc `frequencies` is called for both strings) -> 4N + log N * N